src/backend/distributed/planner/local_plan_cache.c

```diff
@@ -99,6 +99,106 @@ CacheLocalPlanForShardQuery(Task *task, DistributedPlan *originalDistributedPlan
 }
 
 
+/*
+ * GetLocalShardQuery is a helper function which generates the local shard
+ * query based on the jobQuery.
+ *
+ * It is not guaranteed to have consistent attribute numbers on the shards
+ * and on the shell (e.g., distributed/reference tables) due to DROP COLUMN
+ * commands. To avoid any edge cases due to such discrepancies, we first
+ * deparse the jobQuery with the tables replaced to shards, and parse the
+ * query string back. This is normally a very expensive operation, however
+ * we only do it once per cached local plan, which is acceptable.
+ */
+static Query *
+GetLocalShardQuery(Query *jobQuery, Task *task, ParamListInfo orig_paramListInfo)
```

Comment:
We probably should emphasize in the name that this is specifically for local plan caching (at least, as long as we do not implement insert..select) and not a general function for getting a local shard query.

Reply:
changed the name to `GetLocalShardQueryForCache`